### PR TITLE
revert idiosyncrasies select

### LINKED ISSUE #42624: Revert some idiosyncrasies from MaterializedMySQL
**Use case**

When contributors added `MaterializedMySQL` they also allowed some obscure identifiers for MySQL compatibility, for example:

```
SELECT 1abc
```

We should not allow them.
Let's revert the related changes in Lexer.

### PR DESCRIPTION
**Pull request by zzsmdfj**
### Changelog category (leave one):

- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

backward incompatible change：revert idiosyncrasies select . close:  https://github.com/ClickHouse/ClickHouse/issues/42624

### CODE REVIEW

File: src/Parsers/Lexer.cpp [hunk 1]
```diff
@@ -385,10 +378,10 @@ Token Lexer::nextTokenImpl()
                     return Token(TokenType::DollarSign, token_begin, ++pos);
                 }
             }
-            if (isWordCharASCII(*pos) || *pos == '$')
+            if (isWordCharASCII(*pos))
```
Comment (alexey-milovidov) [inline src/Parsers/Lexer.cpp:381]: Although `$` is already gets widely used, let's keep it.

Comment (zzsmdfj) [reply src/Parsers/Lexer.cpp]: done

**Review by serxa — APPROVED** [review]
LGTM, let's see what CI will say

**Comment by serxa** [comment]
Well, yes, looks like getting rid of all "invalid" identifiers in tests is the only way to proceed.

**Comment by tavplubix** [comment]
Please change the changelog category to "backward incompatible change" and write _a user-readable short description of the changes_  ("close: https://github.com/ClickHouse/ClickHouse/issues/42624" is not a changelog entry)

**Comment by zzsmdfj** [comment]
> Well, yes, looks like getting rid of all "invalid" identifiers in tests is the only way to proceed.

I fixed most of the test cases by：
(1)  add ``  
(2) Adaptation identifier, such as:  01053_db -> db_01053

We can adapter most of the test SQL, but I'm a little worried about  compatibility for users when the upgraded version?

**Comment by zzsmdfj** [comment]
@serxa Are there any other faild need to fix or pay attention except  'Stateless tests (aarch64) : 02458_use_structure_from_insertion_table'?  

We need to deal with this pr as soon as possible, or the new tests  cases  will cause continuous tests failure  for pr.

**Comment by serxa** [comment]
`02458_use_structure_from_insertion_table` looks like the last one to fix according to CI. And
 fuzzer complains `AST fuzzer (debug) — Found error: The query formatting is broken.`, I'm not sure if it also related to that test

**Comment by serxa** [comment]
> We can adapter most of the test SQL, but I'm a little worried about compatibility for users when the upgraded version?

@alexey-milovidov Are you totally sure about this "backward incompatible change"? We have a lot of tests rewritten here and also users having `123_my_table` will get in trouble. Don't we need feature flag for it?

**Comment by alexey-milovidov** [comment]
Let's keep MySQL compatibility, it is not too bad.

**Pull request closed without merging**